extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java

```diff
@@ -0,0 +1,386 @@
+   * exception will be recorded.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param runnable the {@link Runnable} to run
+   */
+  public void run(SpanBuilder spanBuilder, Runnable runnable) {
+    call(
+        spanBuilder,
+        () -> {
+          runnable.run();
+          return null;
+        });
+  }
+
+  /**
+   * Runs the given {@link Callable} inside a new span with the given name.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the span will be marked as error, and the
+   * exception will be recorded.
+   *
+   * @param spanName the name of the span
+   * @param callable the {@link Callable} to call
+   * @param <T> the type of the result
+   * @return the result of the {@link Callable}
+   */
+  public <T> T call(String spanName, Callable<T> callable) {
+    return call(tracer.spanBuilder(spanName), callable);
+  }
+
+  /**
+   * Runs the given {@link Callable} inside of the span created by the given {@link SpanBuilder}.
+   * The span will be ended at the end of the {@link Callable}.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the span will be marked as error, and the
+   * exception will be recorded.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param callable the {@link Callable} to call
+   * @param <T> the type of the result
+   * @return the result of the {@link Callable}
+   */
+  public <T> T call(SpanBuilder spanBuilder, Callable<T> callable) {
+    return call(spanBuilder, callable, this::setSpanError);
+  }
+
+  /**
+   * Runs the given {@link Callable} inside of the span created by the given {@link SpanBuilder}.
+   * The span will be ended at the end of the {@link Callable}.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the handleException consumer will be
+   * called, giving you the opportunity to handle the exception and span in a custom way, e.g. not
+   * marking the span as error.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param callable the {@link Callable} to call
+   * @param handleException the consumer to call when an exception is thrown
+   * @param <T> the type of the result
+   * @return the result of the {@link Callable}
+   */
+  @SuppressWarnings("NullAway")
+  public <T> T call(
+      SpanBuilder spanBuilder, Callable<T> callable, BiConsumer<Span, Throwable> handleException) {
+    Span span = spanBuilder.startSpan();
+    //noinspection unused
+    try (Scope unused = span.makeCurrent()) {
+      return callable.call();
+    } catch (Throwable e) {
+      handleException.accept(span, e);
+      sneakyThrow(e);
+      return null;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param description what went wrong
+   */
+  public void setSpanError(Span span, String description) {
+    span.setStatus(StatusCode.ERROR, description);
+  }
```

Comment:
how important is it having the `setSpanError` methods be public? it feels like they are not needed for the primary use cases

Reply:
no - fixed